lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -650,6 +650,45 @@ BlockedEncodingAttr::toLinearLayout(ArrayRef<int64_t> shape) const {
   return combineCtaCgaWithShape(ctaLayout, getCTALayout(), shape);
 }
 
+std::optional<LinearLayout>
+fmaDotToLinearLayout(DotOperandEncodingAttr operandLayout,
+                     ArrayRef<int64_t> shape) {
+  int rank = shape.size();
+  auto blocked = cast<BlockedEncodingAttr>(operandLayout.getParent());
+  MLIRContext *ctx = operandLayout.getContext();
+
+  // TODO: introduce registerOrder or use getOrder(operandLayout)
+  // Currently this order is used in legacy converter, because we do not
+  // have access to full dot operand layout, only parent part.
+  auto regOrder = blocked.getOrder();
+  // TODO: use operandLayout.getThreadOrder()
+  auto threadOrder = blocked.getThreadOrder();
+  auto warpOrder = blocked.getWarpOrder();
+  auto repOrder = blocked.getRepOrder();
+
+  StringAttr kReg = S("register");
+  StringAttr kLane = S("lane");
+  StringAttr kWarp = S("warp");
+
+  SmallVector<unsigned> threadSize = blocked.getSizePerThread();
+  auto kDimIdx = operandLayout.getOpIdx() == 0 ? rank - 1 : rank - 2;
+  threadSize[kDimIdx] = shape[kDimIdx];
+  auto threadShape = blocked.getThreadsPerWarp();
+  auto warpShape = blocked.getWarpsPerCTA();
+
+  SmallVector<StringAttr> repDimNames =
+      permuteDimNames(standardOutDimNames(ctx, rank), repOrder);
+
+  LinearLayout ctaLayout = identityStandardND(kReg, threadSize, regOrder)
+                               .transposeOuts(repDimNames) *
+                           identityStandardND(kLane, threadShape, threadOrder)
+                               .transposeOuts(repDimNames) *
+                           identityStandardND(kWarp, warpShape, warpOrder)
+                               .transposeOuts(repDimNames);
```

Comment:
Is this equivalent to passing `repDimNames` to `reg`, `lane` and `warp` and not transposing?

Reply:
No it is not equvalent, `transposeOuts` reorders names in representation.

Reply:
Let's consider we have 4 warps and two output dimensions ["dim0", "dim1"] with warpOrder = [1, 0].

this will produce following base vectors: [[0, 1], [1, 0]] with assigned dimensions ["dim0", "dim1"].

now, let's consider repOrder is [1, 0] as well, applying `tranposeOuts` function will produce [[1, 0], [0, 1]] with assigned dimensions ["dim1", "dim0"] (pay attention to dim order here, in linear layout representation list of names is an array, not a unordered set).

I perform these transposes on every stage, because currently(at least last time i tried different order, it did not work) you can not multiply layouts with dimensions in different order.